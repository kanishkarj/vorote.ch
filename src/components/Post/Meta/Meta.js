// @flow strict
import React from 'react';
import moment from 'moment';
import styled from 'styled-components';

const PublishedDate = styled.p`
  font-style: italic;
`;

type Props = {
  date: string
};

const Meta = ({ date }: Props) => (
  <div>
    <PublishedDate>Published {moment(date).format('D MMM YYYY')}</PublishedDate>
  </div>
);

export default Meta;
