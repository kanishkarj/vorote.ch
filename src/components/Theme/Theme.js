import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  fonts: {
    body: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;',
    heading: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;',
    monospace: 'Menlo, monospace',
  },
  fontWeights: {
    body: 400,
    heading: 600,
    bold: 700,
  },
  fontSizes: {
    base: 16,
    small: 14,
  },
  lineHeight: 1.625,
  spacing: Math.round(16 * 1.625),
  buttons: {
    height: 35,
    borderRadius: 20,
  },
  breakpoints: {
    xs: 0,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200
  }
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
);

export default Theme;
