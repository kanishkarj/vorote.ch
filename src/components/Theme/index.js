// @flow strict
export { default } from './Theme';
