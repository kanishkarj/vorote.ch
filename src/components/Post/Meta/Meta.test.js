// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Meta from './Meta';
import 'jest-styled-components';
import Theme from '../../Theme';

describe('Meta', () => {
  it('renders correctly', () => {
    const props = {
      date: '2016-09-01'
    };

    const tree = renderer.create(
    <Theme>
      <Meta {...props} />
    </Theme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
