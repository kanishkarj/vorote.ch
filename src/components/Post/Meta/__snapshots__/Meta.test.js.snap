// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meta renders correctly 1`] = `
.c0 {
  font-style: italic;
}

<div>
  <p
    className="c0"
  >
    Published 
    1 Sep 2016
  </p>
</div>
`;
